src/Networking/NexusMods.Networking.GOG/Client.cs

```diff
@@ -0,0 +1,390 @@
+            {
+                { "client_id", ClientId },
+                { "client_secret", ClientSecret },
+                { "grant_type", "refresh_token" },
+                { "refresh_token", authInfo.RefreshToken },
+            };
+        
+            var uri = new Uri(QueryHelpers.AddQueryString(TokenUri.ToString(), tokenQuery));
+        
+            var tokenResponse = await _client.GetFromJsonAsync<TokenResponse>(uri, token);
+            if (tokenResponse == null)
+            {
+                throw new Exception("The OAuth login request did not return a token.");
+            }
+        
+            using var tx = _connection.BeginTransaction();
+            var e = authInfo.Id;
+            tx.Add(e, AuthInfo.AccessToken, tokenResponse.AccessToken);
+            tx.Add(e, AuthInfo.RefreshToken, tokenResponse.RefreshToken);
+            tx.Add(e, AuthInfo.ExpiresAt, DateTimeOffset.UtcNow.AddSeconds(tokenResponse.ExpiresIn));
+            tx.Add(e, AuthInfo.SessionId, tokenResponse.SessionId);
+            tx.Add(e, AuthInfo.UserId, ulong.Parse(tokenResponse.UserId));
+            var result = await tx.Commit();
+            return authInfo.Rebase(result.Db);
+        }
+        finally
+        {
+            _semaphore.Release();
+        }
+    }
+    
+    /// <summary>
+    /// Try to get the authentication information from the database.
+    /// </summary>
+    public bool TryGetAuthInfo(out AuthInfo.ReadOnly authInfo)
+    {
+        foreach (var found in AuthInfo.All(_connection.Db))
+        {
+            authInfo = found;
+            return true;
+        }
+        authInfo = default(AuthInfo.ReadOnly);
+        return false;
+    }
+    
+    /// <summary>
+    /// Get all the builds for a given product and OS.
+    /// </summary>
+    public async Task<Build[]> GetBuilds(ProductId productId, OS os, CancellationToken token)
+    {
+        var msg = await CreateMessage(new Uri($"https://content-system.gog.com/products/{productId}/os/{os}/builds?generation=2"), CancellationToken.None);
+        using var response = await _client.SendAsync(msg, token);
+        if (!response.IsSuccessStatusCode)
+            throw new Exception($"Failed to get builds for {productId} on {os}. {response.StatusCode}");
+
+        await using var responseStream = await response.Content.ReadAsStreamAsync(token);
+        var content = await JsonSerializer.DeserializeAsync<BuildListResponse>(responseStream, _jsonSerializerOptions, token);
+        
+        if (content == null)
+            throw new Exception("Failed to deserialize the builds response.");
+        
+        return content.Items;
+    }
+
+    /// <summary>
+    /// Try to log out the user.
+    /// </summary>
+    public async Task LogOut()
+    {
+        using var tx = _connection.BeginTransaction();
+        foreach (var found in AuthInfo.All(_connection.Db))
+        {
+            tx.Delete(found, false);
+        }
+        await tx.Commit();
```

Comment:
Should the data also be `Excised` after removal, like we do for nexus login data? 
We do both for nexus logins since just excising doesn't trigger DB subscriptions